### PR TITLE
Add --install and --no-install options to phx.new

### PR DESCRIPTION
**Pull request by aaronrenner**
These new options allow the user to skip the "Fetch and install dependencies?" prompt when generating new projects. This option will be very helpful when generating new projects programatically like during integration testing (#3970) or for phoenixdiff.org.

I was a little unsure how to document this pair of options, so I indented them under a "Skipping prompts" heading.

Screenshot of new phx.new options

/cc @josevalim

### CODE REVIEW

File: installer/lib/mix/tasks/phx.new.ex [hunk 1]
```diff
@@ -49,6 +49,9 @@ defmodule Mix.Tasks.Phx.New do
     * `--binary-id` - use `binary_id` as primary key type
       in Ecto schemas
 
+    * `--no-install` - do not fetch and install dependencies
+      while generating this project
```
Comment (josevalim) [inline installer/lib/mix/tasks/phx.new.ex:53]: Given the default is `--install`, we only document the non-default behaviour. :)

Comment (josevalim) [reply installer/lib/mix/tasks/phx.new.ex:53]: Oh, i see though. You are making it so it doesn't even ask. Ignore me, i will add it back.

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Comment by aaronrenner** [comment]
@josevalim Would you mind back-porting this to phoenix 1.5? I'd like to be able to use this in the phx_gen_auth tests without having to run on phoenix master.

**Comment by aaronrenner** [comment]
Thank you! :heart:

**Pull request merged**